### ISSUE TITLE
Persistent context

### ISSUE THREAD

**Issue by darky** [wilk/microjob#47]
Need ability to pass some context firstly and then, it will be always available in workers.
For example, CPU-intensive geo task - check point in polygons.
Polygons so weight and every time serialize - deserialize it so expensive.
Would be better to pass it firstly

```typescript
await job(() => {
}, {persistentCtx: {polygons: [/* many-many polygons */]}});
```

And then on every job execute it always accessible:

```typescript
await job(() => {
  polygons // it accessible here yet.
  
}, {data: {point: [12.3434, 56.3434]}});
```

**Comment by manuel-di-iorio** [comment]
This feature is partially available in my pull request https://github.com/wilk/microjob/pull/41 via the `props` option.

Example:
```js
await job(() => {
  
}, {
  props: { /* arbitrary data */ }
});
```
But this was intended to be used from the external, like:
```js
workers.some(worker => {
    console.log(worker.props);
});
```
Maybe there is the chance to retrieve this object in the job itself without serialization, using `worker.postMessage()`

**Comment by darky** [comment]
https://github.com/wilk/microjob/pull/48 PR

**Comment by wilk** [comment]
@darky Thanks for this issue!

Well, let me check if I got it rightly: you need a global bucket shared between worker threads to avoid multiple massive serialisations/deserialisations, correct?
This could be done with SharedArrayBuffer (shared memory) by you.
However, yes, it could be a useful feature to embed in microjob.

Anyway, your PR is moving the serialisation/deserialisation problem from the user to the core: https://github.com/darky/microjob/commit/67c21aec41ec0ddc3903d6f28cfaae490e41fc95#diff-c9253097723f89dd4716748fab2e00cdR108
Every time the user invokes `job`, the whole `persistentCtx` gets serialised and sent via postMessage and then deserialised from the worker thread.
I think a good solution could be to pass a global shared context from an external facade, convert it to a SharedArrayBuffer and then convert it back with a proper getter from the worker.
I wouldn't use the `job` interface to define a global context: it's ambiguous.

**Comment by darky** [comment]
> Every time the user invokes job, the whole persistentCtx gets serialised and sent via postMessage and then deserialised from the worker thread.

It occurred once at first time, after it always available via https://github.com/darky/microjob/commit/67c21aec41ec0ddc3903d6f28cfaae490e41fc95#diff-5bfbc2def8d97c3939b537c3f6f31b3eR3

> I think a good solution could be to pass a global shared context from an external facade, convert it to a SharedArrayBuffer and then convert it back with a proper getter from the worker.

Can you please provide little example, also you can close #42 via it example :)

**Comment by darky** [comment]
> I wouldn't use the job interface to define a global context: it's ambiguous.

Yep, agree. Maybe better to use `start` function for this purpose?

**Comment by r3wt** [comment]
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏
🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏🙏

**Comment by r3wt** [comment]
> Yep, agree. Maybe better to use start function for this purpose?

In this scenario, would `persistentCtx` be mutable (from within a `job` for example)?

**Comment by darky** [comment]
@r3wt #48 PR can satisfy your needs about mutation
